### PR TITLE
Update patch-Makefile.mac.diff

### PR DESCRIPTION
**Pull request by cave-canem**
###### Type(s)

- [∎] enhancement

###### Tested on

macOS 10.9.5 13F1911 x86_64
Xcode 6.2 6C131e
macOS 10.13.6 17G14042 x86_64
Xcode 10.1 10B61
mp-clang-10 @10.0.1_5
mp-clang-devel @20210401-d1828937_0 (clang version 13.0.0)

###### Verification 
Have you
port lin gptfdisk
--->  Verifying Portfile for gptfdisk
--->  0 errors and 0 warnings found. [checked your Portfile with `port lint`?]
gptfdisk has no tests turned on. [ tried existing tests with `sudo port test`?]
sudo port -vds install [tried a full install with `sudo port -vst install`?]
yes [ tested basic functionality of all binary files?]

### CODE REVIEW

**Comment by reneeotten** [comment]
@cave-canem thanks for the contribution! 

Please do pay attention to the PR checklist and format your commit message according to the guidelines (i.e., `portname: short description`). Also, you state that you performed an install with `sudo port -vst install`; however, the CI fails as it cannot apply the patch - that should likely have happened for you locally as well. Please take a look at the build failure(s), and update the PR accordingly.

**Comment by reneeotten** [comment]
also, PR #11786 should likely be part of this PR as it touches the same port

**Comment by cave-canem** [comment]
Hello Renee Otten!
I am a beginner who also knows very little English.
I built a port of gptfdisk @ 1.0.8_1 (active) and tested it on Mavericks and High Sierra.
I don't know how to combine https://github.com/macports/macports-ports/pull/11785 and https://github.com/macports/macports-ports/pull/11786 into one request.
Could you help me to solve all these problems?
(Could you please correct my requests)?

P.S.
A detailed build log is attached. 
[gptfdisk_main.log.tar.gz](https://github.com/macports/macports-ports/files/6933589/gptfdisk_main.log.tar.gz)

**Comment by reneeotten** [comment]
> Hello Renee Otten!
> I am a beginner who also knows very little English.
> I built a port of gptfdisk @ 1.0.8_1 (active) and tested it on Mavericks and High Sierra.
> I don't know how to combine #11785 and #11786 into one request.
> Could you help me to solve all these problems?
> (Could you please correct my requests)?
> 
> P.S.
> A detailed build log is attached.
> [gptfdisk_main.log.tar.gz](https://github.com/macports/macports-ports/files/6933589/gptfdisk_main.log.tar.gz)

@cave-canem what's the reason you opened to different PRs for a change to one port? In the future, just make the changes in one single PR; if the changes are not too related you can of course have multiple commits in one PR. 

This PR is a branch with the name `patch-3`, the other one is `patch-4`. For now I think the easiest way is to open a terminal, go to wherever you have your local ports tree and do
`git checkout patch-3` --> change to the branch with name "patch-3"
`git rebase patch-4` --> incorporate the changes from branch "patch-4"

now should have both commits present in the branch with name `patch-3`. The last thing is to now combine the two commits into one and change the `revision` to `0`.

To do so, open the Portfile and set `revision` to zero and commit the change `git commit -a --amend`. You can merge the two commits by doing an "interactive rebase"; there are many website explaining this in detail. 
In short, do `git commit rebase -i HEAD~2` change `pick` in the second commit to `fixup` and save. That should ideally have as a result that all your changes are now in one commit. 

Finally, push that change to this PR (`git push origin -f`) assuming that your fork is called `origin` and the `-f-` ("force") is needed because you rewrote the history.
